srcs/app/http/response/delete/HttpDeleteResponse.cpp

```diff
@@ -3,18 +3,64 @@
 HttpDeleteResponse::HttpDeleteResponse(
 	RequestConfig *requestConfig,
 	HttpRequest *request) : HttpBaseResponse(requestConfig, request) {
-	// TODO(any) Implement DELETE
 	if (error_code_) {
-		raw_response_ = HttpErrorResponse(
-										error_code_,
-										request_config_,
-										request_).Content();
+		SetErrorRawResponse_(error_code_);
 	} else {
-		HttpResponse::HeadersMap headers;
-		std::string	body = "Responding to a DELETE request\n";
+		const std::string full_path =
+				request_config_->GetRoot() + request_->GetDecodedPath();
+		try {
+			File file(full_path);
 
-		headers.insert(std::make_pair("Content-Type", "text/plain"));
-		headers.insert(std::make_pair("Location", request_config_->GetReturnUrl()));
-		SetRawResponse_(request_config_->GetReturnStatus(), headers, body);
+			if (file.IsRegularFile() || file.HasEndSlash()) {
+				Delete_(file);
+			} else {
+				SetErrorRawResponse_(409);
+			}
+		} catch(File::Error & e) {
+			SetErrorRawResponse_(e.what());
+		}
 	}
 }
+
+void	HttpDeleteResponse::Delete_(File const &file) {
+	if (file.IsDirectory()) {
+		BuildResponse_(
+			nftw(file.GetPath().c_str(),
+					RemoveSingleFile_,
+					64,
+					FTW_DEPTH | FTW_PHYS));
+	} else {
+		BuildResponse_(
+			RemoveSingleFile_(file.GetPath().c_str(), NULL, 0, NULL));
+	}
+}
+
+int	HttpDeleteResponse::RemoveSingleFile_(
+								const char *fpath,
+								const struct stat *sb,
+								int typeflag,
+								struct FTW *ftwbuf) {
+	// cast to void when function not used as callback
+	if (!sb) (void)sb;
+	if (!typeflag) (void)typeflag;
+	if (!ftwbuf) (void)ftwbuf;
```

Comment:
I don't know, I have mixed feelings about it because you should keep `fpath` which is being used :thinking:. So, having some params named and others don't seems a little bit strange. But I see your point

Reply:
I forgot to keep the `fpath` name :sweat_smile: . For me, it compiles and works correctly. I get your point, and I'm ok with whatever you decide.

Reply:
you could also do it with __attribute__((unused)):
```
diff --git a/srcs/app/http/response/delete/HttpDeleteResponse.cpp b/srcs/app/http/response/delete/HttpDeleteResponse.cpp
index 2443be4..0fcdcd6 100644
--- a/srcs/app/http/response/delete/HttpDeleteResponse.cpp
+++ b/srcs/app/http/response/delete/HttpDeleteResponse.cpp
@@ -36,16 +36,11 @@ void	HttpDeleteResponse::Delete_(File const &file) {
 }
 
 int	HttpDeleteResponse::RemoveSingleFile_(
-								const char *fpath,
-								const struct stat *sb,
-								int typeflag,
-								struct FTW *ftwbuf) {
-	// cast to void when function not used as callback
-	if (!sb) (void)sb;
-	if (!typeflag) (void)typeflag;
-	if (!ftwbuf) (void)ftwbuf;
-
-    return remove(fpath);
+						const char *fpath,
+						__attribute__((unused)) const struct stat *sb,
+						__attribute__((unused)) int typeflag,
+						__attribute__((unused)) struct FTW *ftwbuf) {
+	return remove(fpath);
 }
 
 void	HttpDeleteResponse::BuildResponse_(int return_status) {
```

Reply:
Adding @Damrod suggestion since it is a little bit more explicit and self-explanatory. Both compiled ok!